### DISCUSSION TITLE
Location ID Filter -> Other Location Devices Still there but unavailable

### DISCUSSION THREAD

**Discussion by VeniceNerd** [tsightler/ring-mqtt#492] [Support Q&A]
I just installed this plugin and all devices from my three ring locations showed up. I added the location ID of just the location I want inside my HA and that worked. However, all of the other devices are still showing up as MQTT entities. The just show as unavailable. Is there any way to purge all old MQTT entities?

**Comment by tsightler** [comment]
You'll have to remove them from HA, via the devices or entities screen.  It should be pretty easy to delete them from entities since I believe you can sort on unavailable and select the devices you want to remove.

**Reply by VeniceNerd** [reply]
Ahhh ok cool. Thanks! Was hoping there would be some sort of shortcut to just dump old stuff from MQTT. 

Btw, if I ever rename sensors in the ring app will they be renamed inside HA as well? Or it’ll always be the name when I first set it up?

Amazing work on this add on btw. It was so easy to install!

**Reply by tsightler** [reply]
The name should update if you restart the addon, note that restarting just HA itself is probably not enough as right now the list of devices is only pulled during addon startup, not automatic add/remove of devices (it's on the roadmap, but as I don't have much time to work on this project these days, who knows when full dynamic device support will actually be added, and the code mostly works fine as is).  Also note that, while the display name of the device/entity will update, the entity_id in Home Assistant will stay the same so as not to break any automations or dashboards, but you can manually update the entity_id in HA if you prefer.

For example, I had a dimmable smart-plug that was originally named "Office Outlet", with an entity id of "light.office_outlet".  I moved the plug to my bonus room and renamed it in the Ring App to "Bonus Room Outlet", which, after a restart of the plugin, automatically renamed in the HA UI, however, the entity ID was still "light.office_outlet" until I manually renamed it.

**Reply by VeniceNerd** [reply]
Good to know mate! Device and Entity naming is def a sore spot in HA still. Gets confusing. 

Loving your add on. Got everything working now and will read about about all the details tomorrow to configure everything to my liking. So glad you made this!
